retry.go

```diff
@@ -243,6 +244,12 @@ func (r *retryHTTP) execute() (res *http.Response, err error) {
 						return nil, err
```

Comment:
Given the complexity of this whole error unwrapping from line 235, to improve readability and maintainability, I would recommend wrapping them as a function called:
func unwrapError(err) err
and just change the line 235 to be
```
err = unwrapError(err)
if (err) != nil {
  return nil, err
}

```